File: 605-can-place-flowers/605-can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        
        ans = [0]
        
        for x in flowerbed:
            ans.append(x)
        ans.append(0)
        
        for i in range(1, len(ans)-1):
            if (ans[i-1] == 0 and ans[i] == 0 and ans[i+1] == 0):
                ans[i] = 1
                n -= 1
        
        return n<=0
        

File: 605-can-place-flowers/605-can-place-flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        
        vector<int> ans;
        ans.push_back(0);
        
        for (const auto& x: flowerbed){
            ans.push_back(x);
        }
        ans.push_back(0);
        
        for (int i = 1; i<ans.size()-1; i++){
            if ((ans[i-1] == 0)&&(ans[i]==0)&&(ans[i+1]==0)){
                n--;
                ans[i] = 1;
            }
                
        }
        
        return (n<=0);
    }
};